nixos/modules/services/security/nginx-sso.nix

```
@@ -37,6 +37,11 @@ in {
         nginx-sso configuration
         ([documentation](https://github.com/Luzifer/nginx-sso/wiki/Main-Configuration))
         as a Nix attribute set.
+
+        Options containing secret data should be set to an attribute set
+        containing the attribute `_secret` - a string pointing to a file
+        containing the value the option should be set to. This file should be
+        readable by `nginx-sso`.
```

Please add an example, alone from the description I don't understand it.
```suggestion
        Options containing secret data should be set to an attribute set
        with the attribute `_secret` which has a string assigned which points to a file
        containing the value that should be used in the end.
        This file must be readable by `nginx-sso`.
```

---

I find that somewhat surprising, this is a know pattern, IIRC I had taken this description verbatim from GitLab's module.

I'm not a _big_ fan of your rephrasing, how about:
```suggestion
        Options containing secret data should be set to an attribute set
        with the singleton attribute `_secret` - a string value set to the path
        to the file containing the secret value which should be used in the
        configuration. This file should be readable by `nginx-sso`.
```

---

It should not be readable but nginx-sso, it must be, otherwise it won't work.

---

Yeah I was also going to add an example, don't worry :-)